GVRf/Framework/src/org/gearvrf/GVRMesh.java

```diff
@@ -66,8 +74,8 @@ protected final boolean registerWrapper() {
      *            Array containing the packed vertex data.
      */
     public void setVertices(float[] vertices) {
-        isNonNull(vertices);
-        isValidData(vertices.length, 3);
+        checkValidDataNotVector(INPUT_TYPE_VERTICES, vertices, vertices.length,
```

Comment:
Yes. This will cause a NullPointer problem. The reason I use this kind of implementation is because checkValidDataNotVector is because of the input parameters, there are both char[] and float[] so that I can't tell which is passed in since they are primitive type array. I will figure out another way to handle this.

Thank you.